### ISSUE TITLE
client crash on Hogwarts Legacy launch while Processing Vulkan shaders

### ISSUE THREAD

**Issue by erpadmin** [ValveSoftware/steam-for-linux#10042]
# Compatibility Report
- Name of the game with compatibility issues:
Hogwarts Legacy
- Steam AppID of the game:
990080

## System Information
- GPU: 
RX 6900xt
- Video driver version: 
Mesa 23.1.6
- Kernel version: 
6.1.49-1-MANJARO
- Link to full system information report as [Gist](https://gist.github.com/):
https://gist.github.com/erpadmin/86ef84564bcd9f39d91553c6b90bf169
- Proton version:

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

added this to launch options however a file is not generated

## Symptoms 
steam client exits (crashes?) when vulkan shaders gets to 10%

## Reproduction
launch game, processing vulkan shaders starts, gets to 10% then client crashes

I can avoid the crash if I click *Skip* before it gets to 10%
I have tried deleting the contents of steamapps/compatdata/990080/pfx/drive_c/ProgramData/Hogwarts Legacy

**Comment by kisak-valve** [comment]
Hello @erpadmin, Steam's shader pre-caching subcomponent is handled by the Steam client, not Proton itself, so I've transferred this issue report to the steam-for-linux issue tracker.

There's a known memory leak in mesa 23.1.6 (specifically) which should hopefully be resolved in the next point release of mesa. It's most likely that memory leak which is leading to the kernel OOM killing Steam.

Reference: https://gitlab.freedesktop.org/mesa/mesa/-/issues/9599

**Comment by erpadmin** [comment]
I also have a crash on the Help->Steam Runtime diagnostics (aka Steam System Information) and the journalctl gpu fault errors which seems to correlate to https://github.com/ValveSoftware/steam-for-linux/issues/9267

**Comment by erpadmin** [comment]
oom kill confirmed
```
Sep 03 15:38:59 mypc kernel: oom-kill:constraint=CONSTRAINT_NONE,nodemask=(null),cpuset=user.slice,mems_allowed=0,global_oom,task_memcg=/user.slice/user-1000.slice/user@1000.service/app.slice/app-steam-849ef38d38b042b1a5d685364230c666.scope,task=steamwebhelper,pid=2927,uid=1000
Sep 03 15:38:59 rlm-ms7c94 kernel: Out of memory: Killed process 2927 (steamwebhelper) total-vm:5127660kB, anon-rss:322756kB, file-rss:0kB, shmem-rss:130704kB, UID:1000 pgtables:3244kB oom_score_adj:300

```

**Comment by erpadmin** [comment]
mesa 23.1.7 has released to manjaro stable and fixes this issue and #9267 for me
